### ISSUE TITLE
[suggest feature] support multiple dataviews and add an identifier field to dataview mapping

### ISSUE THREAD

**Issue by samississipi** [Microsoft/PowerBI-visuals#475]
from [DataViewMapping documentation](https://github.com/Microsoft/PowerBI-visuals-docs/blob/master/Capabilities/DataViewMappings.md#dataviewmappings):
> Each valid mapping will produce a DataView, but currently we only support performing one query per visual so in most situations you will only get one DataView. However, you can provide multiple data mappings with different conditions which allow

there seems to be no related timeline.

the support for multiple dataviews would be a very practical feature. for example, I'm now working on a visual has several rendering modes, each of which desires a different query, currently I could only declare the most general dataview mapping and process the raw data myself.

following is the further consideration about this topic.

assuming multiple dataviews is supported some day, since some dataview mappings might be skipped because of condition mismatching, it would be hard to identify each generated dataview just from the index and/or the dataview itself. 

two possible solution:
1. use `null`/`undefined` to place hold index of generated dataviews array if none of conditions of the mapping is met.
2. assign an unique identifier to each dataview mapping and its generated dataview. in this case it would be better to change the type of `dataViewMappings` field of `capabilities` from array to dictionary to avoid duplicated identifiers.

**Comment by spatney** [comment]
agreed, and we are working on both multiple queries and allowing for dynamically changing dataview mappings. And when we do support it we will probably have something like a mapping name, similar to dataroles to identify the correct dataview.
